web/client-api/src/main/java/io/deephaven/web/client/api/widget/plot/Downsample.java

```diff
@@ -0,0 +1,35 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.web.client.api.widget.plot;
+
+import elemental2.promise.Promise;
+import io.deephaven.web.client.api.*;
+import jsinterop.annotations.JsNullable;
+import jsinterop.annotations.JsOptional;
+import jsinterop.annotations.JsType;
+
+/**
+ * Helper class for plot downsampling methods.
+ */
+@JsType(namespace = "dh.plot")
+public class Downsample {
+    /**
+     * Downsamples a table so that the data can be used for a line plot. The downsampled table should have the same
+     * visual fidelity as the original table, but with fewer rows.
+     *
+     * @param table The table to downsample.
+     * @param xCol The name of the X column to downsample. Must be an Instant or long.
+     * @param yCols The names of the Y columns to downsample.
+     * @param width The width of the visible area in pixels.
+     * @param xRange The visible range as `[start, end]` or null to always use all data.
+     * 
+     * @return A promise that resolves to the downsampled table.
+     */
+    public static Promise<JsTable> linearDownsample(JsTable table, String xCol, String[] yCols, int width,
```

Comment:
What do you think about the original name of this operation, runChartDownsample instead? I picked that in trying to find a name for a "line chart, but the line never changes direction on the X axis", and that was the closest I was able to find.

I wouldn't want to imply here that any line chart will do - for example a log X axis, or one that doubles back on itself will definitely be broken. (Note that log Y axis is fine, and the various Y axes can also have different ranges/etc.)

Might also be worth mentioning in docs that gaps from nulls are preserved as well - and would the [`@experimental`](https://typedoc.org/tags/experimental/) modifier be appropriate if we want to reserve the right to tweak this further?

Reply:
Ya I could change to the original name. I'd say experimental is probably unnecessary unless we think that name is something we'd re-use when expanding downsampling options. Instead we can deprecate this when creating a more comprehensive API